runtime/src/bank.rs

```diff
@@ -3205,9 +3262,13 @@ impl Bank {
 
         debug!("simulate_transaction: {:?}", timings);
 
+        let execution_result = execution_results.pop().unwrap();
         TransactionSimulationResult {
-            result,
-            logs,
+            result: execution_result.flattened_result(),
+            logs: execution_result
+                .details()
+                .and_then(|details| details.log_messages.clone())
```

Comment:
Again, can we destructure `execution_result` instead of doing a partial `clone()` here?